codechecker_lib/analyzers/result_handler_plist_to_db.py

```diff
@@ -33,6 +34,133 @@ def __init__(self, buildaction, workspace, run_id):
+                    event.end_pos.line,
+                    event.end_pos.col,
+                    event.msg,
+                    file_ids[event.start_pos.file_path]))
+ 
+            bug_hash = bug.hash_value
+ 
+            severity_name = self.severity_map.get(bug.checker_name,
+                                                  'UNSPECIFIED')
+            severity = shared.ttypes.Severity._NAMES_TO_VALUES[severity_name]
+ 
+            suppress = False
+ 
+            source_file = bug.file_path
+            last_bug_event = bug.events()[-1]
+            bug_line = last_bug_event.start_pos.line
+ 
+            sp_handler = suppress_handler.SourceSuppressHandler(source_file,
+                                                                bug_line)
+            # check for suppress comment
+            supp = sp_handler.check_source_suppress()
+            if supp:
+                # something shoud be suppressed
+                suppress_checkers = sp_handler.suppressed_checkers()
+ 
+                if bug.checker_name in suppress_checkers or \
+                   suppress_checkers == ['all']:
+                    suppress = True
+ 
+                    file_path, file_name = ntpath.split(source_file)
+ 
+                    # checker_hash, file_name, comment
+                    to_suppress = (bug_hash,
+                                   file_name,
+                                   sp_handler.suppress_comment())
+ 
+                    LOG.debug(to_suppress)
+ 
+                    connection.add_suppress_bug(self.__run_id, [to_suppress])
+ 
+            LOG.debug('Storing check results to the database')
+ 
+            report_id = connection.add_report(analisys_id,
+                                              file_ids[bug.file_path],
+                                              bug_hash,
+                                              bug.msg,
+                                              bug_paths,
+                                              bug_events,
+                                              bug.checker_name,
+                                              bug.category,
+                                              bug.type,
+                                              severity,
+                                              suppress)
+ 
+            report_ids.append(report_id)
+
+    def handle_plist(self, plist):
+        with client.get_connection() as connection:
+            analisys_id = connection.add_build_action(self.__run_id,
+                                                      plist,
+                                                      'Build action from plist')
+
+            try:
+                files, bugs = plist_parser.parse_plist(plist)
+            except Exception as ex:
+                msg = 'Parsing the generated result file failed'
+                LOG.error(msg + ' ' + plist)
+                connection.finish_build_action(analysis_id, msg)
+                return 1
```

Comment:
Because as far as I have seen, this is the general way of failure reporting, like in the `except` block of `handle_results()` function.